backend/src/main/java/com/github/bolabolor/backend/service/UserService.java

```diff
@@ -0,0 +1,28 @@
+package com.github.bolabolor.backend.service;
+import com.github.bolabolor.backend.model.MongoUserDTO;
+import com.github.bolabolor.backend.security.MongoUser;
+import com.github.bolabolor.backend.security.MongoUserRepository;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserService {
+    private final MongoUserRepository mongoUserRepository;
+    private final PasswordEncoder encoder;
+
+    public UserService(MongoUserRepository mongoUserRepository, PasswordEncoder encoder) {
+        this.mongoUserRepository = mongoUserRepository;
+        this.encoder = encoder;
+    }
+
+    public MongoUser signupMongoUser(MongoUserDTO mongoUserDTO) {
+
+        if (mongoUserRepository.findMongoUserByUsername(mongoUserDTO.username()).isPresent()) {
+            throw new IllegalArgumentException("The username already exists.");
+        } else {
+            String encodedPassword = encoder.encode(mongoUserDTO.password());
+            MongoUser encodedUser = new MongoUser(mongoUserDTO.username(), encodedPassword);
+            return mongoUserRepository.save(encodedUser);
+        }
```

Comment:
Wenn du in deinen 1. if Block gehst, dann verlässt du nach dem throw new ...Exception die Methode.
Du kannst das else also weg lassen. Falls es für dich so verständlicher ist, auch gut.
```suggestion
        if (mongoUserRepository.findMongoUserByUsername(mongoUserDTO.username()).isPresent()) {
            throw new IllegalArgumentException("The username already exists.");
        }
            
        String encodedPassword = encoder.encode(mongoUserDTO.password());
        MongoUser encodedUser = new MongoUser(mongoUserDTO.username(), encodedPassword);
        return mongoUserRepository.save(encodedUser);
  
```